### PR TITLE
Datasync minor fixes

### LINKED ISSUE #1577: [DataSync] warning if no datasync models found?
Opinion: It would be great if DataSync logged something if it couldn't find any DataSync Models, would definitely make for somewhat easier debugging

### LINKED ISSUE #1583: [DataSync] Add `updatedAt` fields to documents without one (on update)
Relevant discussion in #1570

### PR DESCRIPTION
**Pull request by ssd71**
closes #1583, closes #1577

### CODE REVIEW

File: packages/graphback-datasync/src/DataSyncPlugin.ts [hunk 1]
```diff
@@ -85,6 +87,10 @@ export class DataSyncPlugin extends GraphbackPlugin {
       }
     })
 
+    if (dataSyncModelCount === 0) {
+      console.error("No DataSync Models detected, ensure that your models are properly annotated.")
```
Comment (wtrocki) [inline packages/graphback-datasync/src/DataSyncPlugin.ts:91]:
```suggestion
      console.warn("No DataSync Models detected, ensure that your models are properly annotated.")
```

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
 
     data._deleted = true;
     const objectId = new ObjectId(idField.value);
-    const result = await this.db.collection(this.collectionName).updateOne({ _id: objectId, _deleted: false }, { $set: data });
+    const result = await this.db.collection(this.collectionName).updateOne({ _id: objectId, _deleted: { $ne: true } }, { $set: data });
```
Comment (wtrocki) [inline packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:59]: I do not understand reasoning behind this change. 
first - why we need to have _deleted here - It seems redundant. Also ne: true looks strange

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts [hunk 3]
```diff
@@ -76,7 +76,7 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
     const projection = this.buildProjectionOption(context);
     const query = this.db.collection(this.collectionName).findOne({
       ...filter,
-      _deleted: false
+      _deleted: { $ne: true}
```
Comment (wtrocki) [inline packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:79]: I guess we want to cather situation when `_deleted` doesn't exist?

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts [hunk 4]
```diff
@@ -121,23 +121,23 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
...
       ...this.buildProjectionOption(context),
       [fieldNames.updatedAt]: 1,
       _deleted: 1
     };
-    const queryResult = await this.db.collection(this.collectionName).findOne({ _id: new ObjectId(idField.value), _deleted: false }, { projection });
+    const queryResult = await this.db.collection(this.collectionName).findOne({ _id: new ObjectId(idField.value), _deleted: { $ne: true} }, { projection });
     if (queryResult) {
       queryResult[idField.name] = queryResult._id;
-      if (clientData[fieldNames.updatedAt].toString() !== queryResult[fieldNames.updatedAt].toString()) {
+      if (queryResult[fieldNames.updatedAt] !== undefined && clientData[fieldNames.updatedAt].toString() !== queryResult[fieldNames.updatedAt].toString()) {
```
Comment (wtrocki) [inline packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:134]: Too long statement.

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts [hunk 5]
```diff
@@ -56,7 +56,7 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
 
     data._deleted = true;
```
Comment (wtrocki) [inline packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:57]: ` this.fieldTransformMap[TransformType.UPDATE]` above? Why we doing this?

Comment (ssd71) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:57]: This is to set the updatedAt timestamp

Comment (wtrocki) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:57]: Ah.. Looks fairly generic mapper for each field. We will need to see how this affects performance

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts (hunk 3 above)
Comment (ssd71) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:79]: Yes :+1:

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts (hunk 2 above)
Comment (ssd71) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:59]: Hmm, I agree _deleted is redundant here

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts (hunk 4 above)
Comment (ssd71) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:134]: Yep, I'll take care of that in next push

**Comment by wtrocki** [comment]
Rebase please

**Pull request merged**